File: Strings/1024.cpp
//https://www.urionlinejudge.com.br/judge/en/problems/view/1024
#include<cstdio>
#include<cstring>
#include <cmath>
using namespace std;

int main(){
    int n,i,j,tam,metade;
	char texto[1002];
    scanf("%d",&n);
    getchar();
    for(i=0; i<n; i++){
             gets(texto);
             tam = strlen(texto);
             for(j=0;j<tam; j++){
				if(texto[j] >= 'a' && texto[j] <= 'z')
					texto[j]+=3;
				else if(texto[j] >= 'A' && texto[j] <= 'Z')
					texto[j]+=3;
            }
			if (tam % 2 == 1)
				metade= tam/2;
			else
				metade= (tam/2)-1;
			 for(j=0;j<=metade; j++){
				texto[j]-=1;
			}
		for(j=tam-1; j>=0; j--)
			printf("%c",texto[j]);

		printf("\n");
    }
    return 0;
}

File: Strings/1024.py
# https://www.urionlinejudge.com.br/judge/en/problems/view/1024
from itertools import count, chain


def crypt(text):
    half = len(text) // 2
    reversed_text = reversed(text)
    first_generator = (chr(ord(c) + 3) if c.isalpha() else c for _, c in zip(range(half), reversed_text))
    second_generator = (chr(ord(c) + 2) if c.isalpha() else chr(ord(c) - 1) for _, c in zip(count(half), reversed_text))

    return ''.join(chain(first_generator, second_generator))


def tests():
    assert crypt('Texto #3') == '3# rvzgV'
    assert crypt('abcABC1') == '1FECedc'
    assert crypt('vxpdylY .ph') == 'ks. \\n{frzx'
    assert crypt('vv.xwfxo.fd') == 'gi.r{hyz-xx'


def uri_format():
    n = int(input())
    for i in range(n):
        text = input()
        print(crypt(text))


if __name__ == '__main__':
    tests()
    uri_format()
